code/modules/mob/living/basic/festivus_pole.dm

```diff
@@ -40,15 +40,18 @@
 	///how much charge we give off to cells around us when rubbed
 	var/recharge_value = 75
 
+
 /mob/living/basic/festivus/Initialize(mapload)
 	. = ..()
 	AddComponent(/datum/component/seethrough_mob)
 	var/static/list/death_loot = list(/obj/item/stack/rods)
 	AddElement(/datum/element/death_drops, death_loot)
 	AddComponent(/datum/component/aggro_emote, emote_list = string_list(list("growls")), emote_chance = 20)
-	var/datum/action/cooldown/mob_cooldown/charge_apc/charge_ability = new(src)
-	charge_ability.Grant(src)
-	ai_controller.set_blackboard_key(BB_FESTIVE_APC, charge_ability)
+
+	var/static/list/innate_actions = list(
+		/datum/action/cooldown/mob_cooldown/charge_apc = BB_FESTIVE_APC,
+	)
+	grant_multiple_actions(innate_actions)
```

Comment:
On top of this, I feel like there's something goofy about passing a singular action into "grant_*multiple*_actions". 

Also it feels a bit overkill, like do we really need a static list containing a single typepath for a mob barely spawned once around?

I'd almost suggest splitting the the proc, one being an easy wrapper `create_action_for_key(action_type, key)` that does new + grant + set, then another proc for `create_multiple_actions_for_keys(list/action_to_key)`.

Reply:
i considered that it was wonky naming but i didn't have a good solution for that. i feel like proc overhead would be rather sad in spots where we have to add a shitload of lists constantly like in spiders (even though i didn't touch spiders in this PR because it's just fucked). 

i think the static list is okay here because i have definitely seen rounds with abounds of festivus pole (and other xenobio mobs) be spammed. there's another option i have for the naming (it will be less silly but i don't think you'll like it much more).

Reply:
Basic mob init isn't that hot, even like worst case we spawn less than 100 spiders a round.

I'm just wary about this because statically cached list is like the biggest fuck-up point of elements (connect loc, death drops) and we're taking rather-user-friendly code and making it less so. Especially since it's like, not even needed. The list is one item long.

Reply:
also for the record, there's still like 10+ cases I didnt' touch where we just do the "create a new `/datum/action` and `Grant()` it" that doesn't use the new proc because it was only one instance. the only reason why this is converted to the proc system was because of the blackboard key. we could convert everything over but i figured it wasn't necessary.

Reply:
> Basic mob init isn't that hot, even like worst case we spawn less than 100 spiders a round.
> 
> I'm just wary about this because statically cached list is like the biggest fuck-up point of elements (connect loc, death drops) and we're taking rather-user-friendly code and making it less so. Especially since it's like, not even needed. The list is one item long.

i get connect loc being a fucked static list but `death_drops` element is a very, very standard application of making and passing static lists. we don't even need the list to be static i just did it like that because someone is going to go " :nerd_face: erm shouldn't these lists be static???"

Reply:
Also I feel like mentioning, we do `add_traits(list(BLAH, BLAH)...)` all the time in basic mob init without statically caching them, and that's pretty much the same thing (proc call, 1 list). 

I think we could drop a ton of boilerplate from this pr by just embracing `funnyproc(list(action, action))`.

Reply:
i think the single lining of some stuff like this would work in certain cases (like this one) so i'm okay with that- but no one is going to learn about static cacheing stuff (which is used in `GLOB` and several other spots) if we keep hand holding this.